casper/src/main/scala/coop/rchain/casper/engine/BlockApproverProtocol.scala

```diff
@@ -166,19 +166,21 @@ object BlockApproverProtocol {
             Long.MaxValue
           )
           .toSet
-        blockDeploys = block.body.deploys
+        blockDeploys       = block.body.deploys
+        blockSystemDeploys = block.body.systemDeploys
         _ <- (blockDeploys.size == genesisBlessedContracts.size)
               .either(())
               .or("Mismatch between number of candidate deploys and expected number of deploys.")
-      } yield (blockDeploys, block.body.state)
+      } yield (blockDeploys, blockSystemDeploys, block.body.state)
 
     (for {
-      result                    <- EitherT(validate.pure[F])
-      (blockDeploys, postState) = result
+      result                                        <- EitherT(validate.pure[F])
+      (blockDeploys, blockSystemDeploys, postState) = result
       stateHash <- EitherT(
                     runtimeManager
                       .replayComputeState(runtimeManager.emptyStateHash)(
                         blockDeploys,
+                        blockSystemDeploys,
```

Comment:
Because the pre-existing `replaySystemDeploy(startHash: StateHash)(...)` function does not fit well into `RuntimeManager.replayComputeState` which is unaware of the the start state hashes of each deploy or system deploy.